### PR TITLE
fix(DIA-815): add AuthImpression and ResetYourPassword events (and more!)

### PR DESCRIPTION
**Pull request by iskounen**
This PR resolves [DIA-815]

### Description

- [x] Add "tap" to `AuthTrigger` schema
- [x] Add `AuthImpression` event
- [x] Add `trigger` property to `AuthImpression`
- [x] Add `ResetYourPassword` event
- [x] "Password reset link set—check your email. Please note, you must wait 5 minutes to receive another link."
- [x] "Get Artsy’s emails on the art market, products, services, editorial, and promotional content. Unsubscribe at any time."
- [ ] ~Determine whether we can surmise a signup or login from the social sign-in service~
- [ ] ~Handle TODO in onSocialLogin~

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Add AuthImpression and ResetYourPassword events to new auth flow - iskounen

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-815]: https://artsyproduct.atlassian.net/browse/DIA-815?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Onboarding/Auth2/components/AuthModal.tsx [hunk 1]
```diff
@@ -74,3 +82,10 @@ export const AuthModal: React.FC = ({ children }) => {
     // </KeyboardAvoidingView>
   )
 }
+
+const tracks = {
```
Comment (damassi) [inline src/app/Scenes/Onboarding/Auth2/components/AuthModal.tsx:86]: @iskounen - we've been migrating away from this pattern across all of our codebases in favor of something a little easier to work with, test etc. 

For any given "app", we should create an associated `useAppNameTracking` hook. And then in there we drop all the tracking calls in a form similar to
- https://github.com/artsy/eigen/blob/main/src/app/Scenes/HomeView/useHomeViewTracking.ts. 

By grouping in this way, we can consolidate all tracking behavior in one thing, which keeps it tidy and easier to maintain through time. And From there, test wise, we can simplify: 

At call sites in code can do something like

```tsx
jest.mock('path/to/useAppNameTracking')

... 
const mockUseAppNameTracking = jest.fn() as jest.Mock
... 

it('tracks clicks', () => {
  const spy = jest.fn() 
  mockUseAppNameTracking.mockReturnValue({ trackSomeThing: spy })
  
  thing.fireEvent('click') 
  expect(spy).toHaveBeenCalled() 
}) 
```

And then in the `useAppNameTracking` file, the tests can test actual properties by simply invoking each of the tracking functions individually and asserting the return values.

File: src/app/Scenes/Onboarding/Auth2/scenes/ForgotPasswordStep.tsx [hunk 2]
```diff
@@ -28,9 +31,10 @@ export const ForgotPasswordStep: React.FC = () => {
         if (!res) {
           setErrors({
             email:
-              "Couldn’t send reset password link. Please try again, or contact support@artsy.net",
+              "Couldn't send reset password link. Please try again, or contact support@artsy.net",
           })
         } else {
+          tracking.trackEvent(tracks.resetYourPassword())
```
Comment (damassi) [inline src/app/Scenes/Onboarding/Auth2/scenes/ForgotPasswordStep.tsx:37]: By rolling tracking up in this way we're constraining all of the different APIs into one function call:

```tsx
const tracking = useAppNameTracking()
...
onClick={() => tracking.resetYourPassword()}
```

File: src/app/Scenes/Onboarding/Auth2/components/AuthModal.tsx (hunk 1 above)
Comment (iskounen) [reply src/app/Scenes/Onboarding/Auth2/components/AuthModal.tsx:86]: I thought `tracks` was the latest pattern, so thanks for clarifying that. There are a few tracking calls in [./src/app/store/AuthModel.ts](https://github.com/artsy/eigen/blob/e32d8121fb909439365e7fb440ca124a2e8505e4/src/app/store/AuthModel.ts#L874-L883) that I want to switch to this pattern. What would you recommend for the hook name for those?

Comment (damassi) [reply src/app/Scenes/Onboarding/Auth2/components/AuthModal.tsx]: In that case, we just leave it alone -- the hook should only be used within the scope of a given "app"

File: src/app/Scenes/Onboarding/Auth2/scenes/ForgotPasswordStep.tsx (hunk 2 above)
Comment (iskounen) [reply src/app/Scenes/Onboarding/Auth2/scenes/ForgotPasswordStep.tsx]: I added the `AuthImpression` event to the tracking hook, but I decided to move this tracking event out of the Form and put it into the AuthModel handler (where the `CreatedAccount` and `SuccessfullyLoggedIn` events are fired) where it should have been this whole time.

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Add AuthImpression and ResetYourPassword events to new auth flow - iskounen)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 71dd70e8daef8bfca2ec3bf1cefbb215033480a7
</p>

**Pull request merged**